src/verify.c

```diff
@@ -1153,6 +1180,30 @@ enum swupd_code verify_main(void)
 	 * naming convention: All exit goto labels must follow the "brick_the_system_and_FOO:" pattern
 	 */
 
+	/* track bundles specified by user */
+	if (cmdline_option_install && cmdline_option_bundles) {
+		/* this is a fresh install so the tracking directory
+		 * needs to be created */
+		char *new_os_statedir = mk_full_filename(globals.path_prefix, "/var/lib/swupd");
```

Comment:
I gave this a lot of thought, and actually the first implementation I wrote used the statedir but decided to change that. 

This is why I think this is better.

When using os-install to install a system, the only required argument is the path where the OS will be installed, as shown here:
`$ swupd os-install /my/installation/path`

Since we are not specifying a statedir, swupd will use the system's default one, which means it will use the one from the system we are using to install, therefore the tracking file will be created in the statedir of that system, which doesn't make sense cause we are installing the OS in a different location. This leads to two problems: 
1) now the system we are using to install will think it has those bundles installed and tracked (and it may not) 
2) the system that does have those bundles installed won't have those bundles as tracked.

Another alternative is to use the statedir specified by the user. This approach has two other problems: 
1) it would mean os-install would really require the --statedir flag to work correctly, this is a problem because the --statedir is optional, we cannot force the user to use an optional flag. 
2) If the user points to the new statedir in the new system using --statedir, this directory will be empty since it is a new system, so the user will never be able to use the content that may already exist in the statedir of the system from which he is installing.

Reply:
You have a point here. I think that as I said both solutions have problems, but it looks like saving it to path_prefix is better.